StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -0,0 +1,378 @@
+        let label = PaymentSheetUI.makeHeaderLabel(appearance: appearance)
+        label.text = viewModel.header
+        return label
+    }()
+
+    private lazy var updateButton: ConfirmButton = {
+        let button = ConfirmButton(state: .disabled, callToAction: .custom(title: .Localized.save), appearance: appearance, didTap: {  [weak self] in
+            switch self?.viewModel.paymentMethod.type {
+            case .card:
+                Task {
+                    await self?.updateCard()
+                }
+            default:
+                fatalError("Updating payment method has not been implemented for \(self?.viewModel.paymentMethod.type ?? .unknown)")
+            }
+        })
+        button.isHidden = !viewModel.canEdit
+        return button
+    }()
+
+    private lazy var removeButton: UIButton = {
+        let button = UIButton(type: .custom)
+        if #available(iOS 15.0, *) {
+            var configuration = UIButton.Configuration.bordered()
+            configuration.contentInsets = NSDirectionalEdgeInsets(top: 10, leading: 16, bottom: 10, trailing: 16)
+            configuration.baseBackgroundColor = .clear
+            button.configuration = configuration
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 10, left: 16, bottom: 10, right: 16)
+        }
+        button.setTitleColor(appearance.colors.danger, for: .normal)
+        button.layer.borderColor = appearance.colors.danger.cgColor
+        button.layer.borderWidth = appearance.primaryButton.borderWidth
+        button.layer.cornerRadius = appearance.cornerRadius
+        button.setTitle(.Localized.remove, for: .normal)
+        button.titleLabel?.textAlignment = .center
+        button.titleLabel?.font = appearance.scaledFont(for: appearance.font.base.medium, style: .callout, maximumPointSize: 25)
+        button.titleLabel?.adjustsFontForContentSizeCategory = true
+        button.addTarget(self, action: #selector(removePaymentMethod), for: .touchUpInside)
+        button.isHidden = !viewModel.canRemove
+        return button
+    }()
+
+    private lazy var paymentMethodDetails: UIStackView = {
+        lazy var details: UIView = {
+            switch viewModel.paymentMethod.type {
+            case .card:
+                return cardSection.view
+            case .USBankAccount:
+                return usBankAccountSection
+            case .SEPADebit:
+                return sepaDebitSection
+            default:
+                fatalError("Updating payment method has not been implemented for \(viewModel.paymentMethod.type)")
+            }
+        }()
+        let paymentMethodDetails = UIStackView(arrangedSubviews: [details, detailsCannotBeChangedLabel])
+        paymentMethodDetails.axis = .vertical
+        paymentMethodDetails.setCustomSpacing(8, after: details) // custom spacing from figma
+        return paymentMethodDetails
+    }()
+
+    private lazy var detailsCannotBeChangedLabel: UITextView = {
```

Comment:
In the future I think this could say something else like only card brand can be updated. Maybe name this "footnoteLabel" or something more generic.

Reply:
Let's make this optional, and instead of hiding it return nil if we don't need to show it. Minor performance bump. We can check the cases where we set hidden to true before creating the label and return nil if we need to.

Reply:
Done and done